athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```diff
@@ -0,0 +1,385 @@
+                //If VARCHAR
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("Varchar")) {
+                    tableSchemaBuilder.addStringField(definition.getString("COLUMN_NAME"));
+                }
+                //If Numeric
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("NUMERIC")) {
+                    tableSchemaBuilder.addDecimalField(definition.getString("COLUMN_NAME"), 10, 2);
+                }
+            }
+
+        }
+        catch (SQLException e) {
+           logger.info("SQL Exception in doGetTable : "+ e.getMessage());
+        }
+        return new GetTableResponse(request.getCatalogName(),
+                request.getTableName(),
+                tableSchemaBuilder.build(),
+                partitionCols
+                );
+    }
+
+    /**
+     * Here we inject the additional column to hold the Prepared SQL Statement.
+     *
+     * @param partitionSchemaBuilder The SchemaBuilder you can use to add additional columns and metadata to the
+     * partitions response.
+     * @param request The GetTableLayoutResquest that triggered this call.
+     */
+    @Override
+    public void enhancePartitionSchema(SchemaBuilder partitionSchemaBuilder, GetTableLayoutRequest request) {
+
+        logger.info("{}: Catalog {}, table {}", request.getQueryId(), request.getTableName().getSchemaName(), request.getTableName());
+        partitionSchemaBuilder.addField("preparedStmt", new ArrowType.Utf8());
+    }
+
+    /**
+     * Used to get the partitions that must be read from the request table in order to satisfy the requested predicate.
+     * Here generating the SQL from the request and attaching it as a additional column
+     *
+     * @param blockWriter Used to write rows (partitions) into the Apache Arrow response.
+     * @param request Provides details of the catalog, database, and table being queried as well as any filter predicate.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing work for a query that has already terminated
+     */
+    @Override
+        public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request, QueryStatusChecker queryStatusChecker)
+            throws Exception
+    {
+        logger.info("in getPartitions: "+ request);
+
+        StringBuilder sqlStmt = new StringBuilder();
+        Schema schemaName = request.getSchema();
+        TableName tableName = request.getTableName();
+        Constraints contr  = request.getConstraints();
+
+        try {
+            Connection connection = getOrCreateConn(request);
+
+            //Generate the SQL from user provided query
+            String colNames = schemaName.getFields().stream().map(Field::getName).collect(Collectors.joining(","));
+            sqlStmt.append(colNames);
+
+            String preparedStmt = verticaJdbcSplitQueryBuilder.buildSql(tableName.getSchemaName(), tableName.getTableName(), schemaName, contr, request.getQueryId());
```

Comment:
This line is a bit long, can you format it as follows:

```java
String preparedStmt = verticaJdbcSplitQueryBuilder.buildSql(tableName.getSchemaName(), 
                                                            tableName.getTableName(), 
                                                            schemaName, 
                                                            contr, 
                                                            request.getQueryId());
```

Reply:
done